portfolio/src/main/webapp/script.js

```diff
@@ -37,3 +37,50 @@ window.addEventListener("scroll", function() {
   }
 });
 
+async function getComments() {
+  console.log("in get comments");
+  const numComments = document.getElementById("numComments").value;
+  const response = await fetch('/data?numComments='+ numComments);
```

Comment:
What you are doing here with the await function is blocking the current thread until the fetch function finishes. It's removing the async logic from this code. You should instead provide a "callback function" that should execute once this operation completes. You set a callback with the `.then()` function. You can reorganize your code into something like
```
fetch('/data?numComments='+ numComments)
  .then(response => response.json())
  .then((comments) => {
    const commentSection = document.getElementById('comment-list');
    commentSection.innerText = ""; // clear old comments 
    comments.forEach((comment) => {
      commentSection.appendChild(createCommentElement(comment));
    }
);

```

Also check out this page for another example of using promises
https://github.com/googleinterns/step/blob/master/walkthroughs/week-3-server/examples/subtraction-game/src/main/webapp/script.js#L19